ir/base.def

```diff
@@ -445,6 +445,12 @@ interface IAnnotated {
     inline void addOrReplaceAnnotation(Annotation ann) {
         Annotations::addOrReplace(getAnnotations(), ann);
     }
+#emit
+    // FIXME: We shouldn't need this delete.
+    // Ideally, we only accept references to expressions.
+    void addOrReplaceAnnotation(cstring name, std::nullptr_t expr, bool structured = false) = delete;
+    void addAnnotation(cstring name, std::nullptr_t expr, bool structured = false) = delete;
```

Comment:
While calling these with an explicit `nullptr` doesn't make sense, does this do anything for the `{}` case?  I think what we need is something like:

    inline void addOrReplaceAnnotation(cstring name, std::initializer_list<const Expression *> elist, bool structured = false) {
        Annotations::addOrReplace(getAnnotations(), new Annotation(name, elist, structured));
    }

or perhaps add `Annotations::addOrReplace(Vector<Annotation> &, cstring, std::initializer_list<const Expression *>, bool)` and call that.

Reply:
Deleting this overload works because the compiler will throw an error that the call is ambiguous. It is not pretty, but doesn't break the interface. 

Probably the right approach is to make everything pass-by-reference.